src/lib/bcmwl/src/bcmwl_vap.c

```diff
@@ -537,7 +537,12 @@ void bcmwl_vap_prealloc_all(void)
             bssmax = bcmwl_radio_max_vifs(p->d_name);
             if (bssmax > BCMWL_VAP_PREALLOC_MAX)
                 bssmax = BCMWL_VAP_PREALLOC_MAX;
-            bcmwl_vap_prealloc(p->d_name, bssmax - 1, bcmwl_vap_mac_xfrm);
+            if(bssmax > 8){
+                    bcmwl_vap_prealloc(p->d_name, bssmax - 1, bcmwl_vap_mac_xfrm);
+            }else{
+		    //Adding a log here would be helpful to inform the developer/integrator
+                    bcmwl_vap_prealloc(p->d_name, bssmax - 2, bcmwl_vap_mac_xfrm);
+            }
```

Comment:
1. Formatting does not match the existing style. This needs to be fixed.
2. bssmax will never be greater than 8 by default - the BCMWL_VAP_PREALLOC_MAX is already 8, and will limit bssmax to be 8 at most. Are you building opensync with modified BCMWL_VAP_PREALLOC_MAX? If so, to what value, and why?
3. Why are you checking bssmax against 8 in particular? Why not 7 or 9?
4. Why are you decrementing 1 or 2 depending on the condition result? **How** is this helping, and **why** is this helping with "Wi-Fi Association Bug". 
5. What is the _bug_ that you're really fixing? Can you describe what you _do_, and what you _see_, and what you _expect_?